src/iperf_udp.c

```diff
@@ -556,26 +556,49 @@ iperf_udp_connect(struct iperf_test *test)
     setsockopt(s, SOL_SOCKET, SO_RCVTIMEO, (struct timeval *)&tv, sizeof(struct timeval));
 #endif
 
-    /*
-     * Write a datagram to the UDP stream to let the server know we're here.
-     * The server learns our address by obtaining its peer's address.
-     */
-    buf = 123456789;		/* this can be pretty much anything */
-    if (write(s, &buf, sizeof(buf)) < 0) {
-        // XXX: Should this be changed to IESTREAMCONNECT? 
-        i_errno = IESTREAMWRITE;
-        return -1;
-    }
+    for (int i=0; i<5; i++){
+        /* retry the connection setup for up to 5 times */
+
+        /*
+         * Write a datagram to the UDP stream to let the server know we're here.
+         * The server learns our address by obtaining its peer's address.
+         */
+        buf = 123456789;		/* this can be pretty much anything */
```

Comment:
Once we are running multiple processes of iperf, they will get the same buf for each process.
i would like to suggest adding +i to the buffer will solve the issue.